src/rule-parser.js

```diff
@@ -70,17 +69,62 @@ class Requirements {
     case '!=':
       return actualAmount !== number;
     default:
-      console.error(`Unexpected operator : ${operator}`);
+      throw new Error(`Unexpected operator : ${operator}`);
     }
   }
 }
 
-try {
-  var rules = yaml.safeLoad(fs.readFileSync('./config/gamification.yml', 'utf8'));
-} catch (e) {
-  console.error(e);
+class XPRequirement extends Requirement {
+  constructor(requirement) {
+    requirement['amount'] = requirement['amount'] === undefined ? 1 : requirement['amount'];
+    super(requirement);
+  }
+
+  async isFulfilled(context) {
+    const matches = await context.app.service('xp').find({
+      query: {
+        user_id: context.data.user_id,
+        name: this.requirement['xp']
+      }
+    });
+    if (matches.length === 0) return false;
+    if (matches.length > 1) throw new Error('Found more than one match, this should be impossible');
+    if (!Requirement.isValidAmount(matches[0].amount, this.requirement['amount'])) return false;
+
+    return true;
+  }
 }
 
+class AchievementRequirement extends Requirement {
+  constructor(requirement) {
+    requirement['amount'] = requirement['amount'] === undefined ? 1 : requirement['amount'];
+    super(requirement);
+  }
+
+  async isFulfilled(context) {
+    const matches = await context.app.service('achievements').find({
+      query: {
+        user_id: context.data.user_id,
+        name: this.requirement['achievement']
+      }
+    });
+    if (matches.length === 0) return false;
+    if (matches.length > 1) throw new Error('Found more than one match, this should be impossible');
+    if (!Requirement.isValidAmount(matches[0].amount, this.requirement['amount'])) return false;
+
+    return true;
+  }
+}
+
+// TODO perhaps add conditions, events etc for requirements
+class EventRequirement extends  Requirement {
```

Comment:
We will also need an amount check in the constuctor here. But could also add it when implementing the whole logic.